src/utils.ts

```diff
@@ -48,6 +48,29 @@ export const promiseWithTimeout = <T>(
   })
 }
 
+// Implements type safe retries of arbitrary async functions
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function retry<T extends (...arg0: any[]) => any>(
+  fn: T,
+  args: Parameters<T>,
+  maxRetries: number,
+  sleepTime: number,
+  retryCount = 1
+): Promise<Awaited<ReturnType<T>>> {
+  const currRetry = typeof retryCount === 'number' ? retryCount : 1
+  try {
+    const result = await fn(...args)
+    return result
+  } catch (e) {
+    console.error(e)
```

Comment:
Should we just `log` or `debug` here instead?

Reply:
I can switch to a log